pkg/store/disk.go

```diff
@@ -0,0 +1,71 @@
+package store
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/lima-vm/lima/pkg/qemu/imgutil"
+	"github.com/lima-vm/lima/pkg/store/filenames"
+)
+
+type Disk struct {
+	Name        string `json:"name"`
+	Size        int64  `json:"size"`
+	Dir         string `json:"dir"`
+	Instance    string `json:"instance"`
+	InstanceDir string `json:"instanceDir"`
+	MountPoint  string `json:"mountPoint"`
+}
+
+func InspectDisk(diskName string) (*Disk, error) {
+	disk := &Disk{
+		Name: diskName,
+	}
+
+	diskDir, err := DiskDir(diskName)
+	if err != nil {
+		return nil, err
+	}
+
+	disk.Dir = diskDir
+	dataDisk := filepath.Join(diskDir, filenames.DataDisk)
+	if _, err := os.Stat(dataDisk); err != nil {
+		return nil, err
+	}
+
+	info, err := imgutil.GetInfo(dataDisk)
+	if err != nil {
+		return nil, err
+	}
+	disk.Size = info.VSize
+
+	instDir, err := os.Readlink(filepath.Join(diskDir, filenames.InUseBy))
+	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			disk.Instance = ""
+			disk.InstanceDir = ""
+		} else {
+			return nil, err
+		}
```

Comment:
The struct is already initialized with the null values, so the explicit assignment seems redundant:

```suggestion
		if !errors.Is(err, fs.ErrNotExist) {
			return nil, err
		}
```